pkg/trace/agent/fuzz_test.go

```diff
@@ -0,0 +1,121 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build go1.18
+
+package agent
+
+import (
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+	"github.com/DataDog/datadog-agent/pkg/trace/testutil"
+)
+
+func FuzzProcessStats(f *testing.F) {
+	agent, cancel := agentWithDefaults()
+	defer cancel()
+	encode := func(pbStats pb.ClientStatsPayload) ([]byte, error) {
+		return pbStats.Marshal()
+	}
+	decode := func(stats []byte) (pb.ClientStatsPayload, error) {
+		var payload pb.ClientStatsPayload
+		err := payload.Unmarshal(stats)
+		return payload, err
+	}
+	pbStats := testutil.StatsPayloadSample()
+	stats, err := encode(pbStats)
+	if err != nil {
+		f.Fatal("Couldn't generate seed corpus:", err)
+	}
+	f.Add(stats, "java", "v1")
+	f.Fuzz(func(t *testing.T, stats []byte, lang, version string) {
+		pbStats, err := decode(stats)
+		if err != nil {
+			t.Skip("Skipping invalid payload:", err)
+		}
+		processedStats := agent.processStats(pbStats, lang, version)
+		if _, err = encode(processedStats); err != nil {
+			t.Fatal("processStats returned an invalid stats payload:", err)
+		}
```

Comment:
For additional testing on this, you can encode it, then encode `pbStats` again and make sure it is the same as `stats`. Then decode it once again and make sure that is equal to the originally decoded value. Just to make sure that the encoding/decoding is always consistent.

Reply:
Hmm looks like some non-empty valid inputs make the decode function return an empty `pbStats` and a `nil` error, so `encode(pbStats)` and  `stats` won't be the same. Not sure if it should be considered a bug or an intended behavior in the protobuf marshaller/unmarshaller.
However, I added extra encoding/decoding verifications after processing.

Reply:
> some non-empty valid inputs make the decode function return an empty pbStats and a nil error

That seems like a bug to me. At the very least, let's file an issue to discuss it. (Doesn't need to block this PR though)

Reply:
Created `AIT-4016` to track it!